astropy/modeling/tests/test_functional_models_quantities.py

```diff
@@ -0,0 +1,244 @@
+                'y_0': 2 * u.arcsec, 'r_eff': 2 * u.arcsec, 'n': 4},
+ 'evaluation':[(3 * u.arcsec, 2.5 * u.arcsec, 2.829990489 * u.MJy/u.sr)],
+ 'bounding_box': False},
+]
+
+MODELS = MODELS_1D + MODELS_2D
+
+SCIPY_MODELS = set([Sersic1D, Sersic2D, AiryDisk2D])
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_evaluatate_without_units(model):
+    if not HAS_SCIPY and model['class'] in SCIPY_MODELS:
+        pytest.skip()
+    m = model['class'](**model['parameters'])
+    for args in model['evaluation']:
+        if len(args) == 2:
+            kwargs = OrderedDict(zip(('x', 'y'), args))
+        else:
+            kwargs = OrderedDict(zip(('x', 'y', 'z'), args))
+            if kwargs['x'].unit.is_equivalent(kwargs['y'].unit):
+                kwargs['x'] = kwargs['x'].to(kwargs['y'].unit)
+        mnu = m.without_units_for_data(**kwargs)
+        args = [x.value for x in kwargs.values()]
+        assert_quantity_allclose(mnu(*args[:-1]), args[-1])
+
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_evaluatate_with_units(model):
+    if not HAS_SCIPY and model['class'] in SCIPY_MODELS:
+        pytest.skip()
+    m = model['class'](**model['parameters'])
+    for args in model['evaluation']:
+        assert_quantity_allclose(m(*args[:-1]), args[-1])
+
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_evaluatate_with_units_x_array(model):
+
+    if not HAS_SCIPY and model['class'] in SCIPY_MODELS:
+        pytest.skip()
+
+    m = model['class'](**model['parameters'])
+
+    for args in model['evaluation']:
+        if len(args) == 2:
+            x, y = args
+            x_arr = u.Quantity([x, x])
+            result = m(x_arr)
+            assert_quantity_allclose(result, u.Quantity([y, y]))
+        else:
+            x, y, z = args
+            x_arr = u.Quantity([x, x])
+            y_arr = u.Quantity([y, y])
+            result = m(x_arr, y_arr)
+            assert_quantity_allclose(result, u.Quantity([z, z]))
+
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_evaluatate_with_units_param_array(model):
+
+    if not HAS_SCIPY and model['class'] in SCIPY_MODELS:
+        pytest.skip()
+
+    params = {}
+    for key, value in model['parameters'].items():
+        if value is None:
+            params[key] = None
+        else:
+            params[key] = np.repeat(value, 2)
+
+    m = model['class'](**params)
+    for args in model['evaluation']:
+        if len(args) == 2:
+            x, y = args
+            x_arr = u.Quantity([x, x])
+            result = m(x_arr)
+            assert_quantity_allclose(result, u.Quantity([y, y]))
+        else:
+            x, y, z = args
+            x_arr = u.Quantity([x, x])
+            y_arr = u.Quantity([y, y])
+            result = m(x_arr, y_arr)
+            assert_quantity_allclose(result, u.Quantity([z, z]))
+
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_bounding_box(model):
```

Comment:
Because for some models, bounding_box crashed if units were present, so I had to fix those and wanted a regression test. I agree we could check the actual values, but the purpose here is first and foremost to check that it does anything at all.